src/lib/tuning/tuning_choice.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include <memory>
+#include <set>
+
+#include "tuning/tuning_operation.hpp"
+
+namespace opossum {
+
+/**
+ * A TuningChoice represents a possible system modification with a certain
+ * performance impact coming at a cost.
+ *
+ * TuningChoices are generated by AbstractEvaluators in a Tuner instance
+ * and either accepted or rejected by an AbstractSelector while generating
+ * a sequence of TuningOperations to improve the overall system performance.
+ */
+class TuningChoice {
+ public:
+  virtual ~TuningChoice() {}
+  /**
+   * An estimate regarding the performance impact of this modification.
+   * values < 0.0f: expected degradation of system performance
+   * values > 0.0f: expected improvement of system performance
+   *
+   * No scaling constraints are specified so that this value is only comparable
+   * among TuningChoices generated by compatible AbstractEvaluator implementations.
+   */
+  virtual float desirability() const = 0;
+
+  /**
+   * How certain the producing evaluator was when generating this choice.
+   * Very basic evaluators will probably output choices with low confidence
+   * in general.
+   * Similarly, if an evaluator is very specialized for a certain edge case,
+   * it might produce few choices with high confidence and several other
+   * choices with a low confidence value, as other evaluators might be able
+   * to give better results in these cases.
+   */
+  virtual float confidence() const = 0;
```

Comment:
You don’t seem to use this anywhere. Since this would be easy to add, I would leave it out for now and iif someone needs it in the future, they can put it in themselves. For me, this seems better than pretending it exists when we just have a stub.

Reply:
We added this in agreement with @Bensk1 but see your point. If you two agree, we will remove it.